### PR TITLE
Video Component Controls

### PR DESCRIPTION
**Pull request by yichenshen**
Port over video controls from what Wudi has.

The player is altered from the original to wrap around react-player instead, so videos from other sources can be considered at a later time too. The slider is redone too, since the original was a modification on a cloned copy of material-ui/slider, and is quite hard to maintain. 

Currently, the states are kept in VideoPlayer itself but will be moved into redux in subsequent iterations.

The component is not enabled as of yet as it is WIP. Enable in `app/controllers/components/course/videos_component.rb` to test.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -91,6 +92,7 @@
     "react-dropzone": "^3.13.2",
     "react-intl": "^2.3.0",
     "react-overlays": "^0.7.0",
+    "react-player": "^0.24.2",
```
Comment (allenwq) [inline client/package.json]: We should make these libraries async if they're only used in the video page. That is keep these two libs separete from the main bundle and only load then on video page. You can check webpack documentation for async import, and there are some examples in the existing code too.

Comment (yichenshen) [reply client/package.json]: Alright, that sounds like a better way to do it, yes. Will look into ti!

**Comment by yichenshen** [comment]
@allenwq 
I wasn't able to find a specific example that loaded react components dynamically. But short of importing packages like `react-code-splitting`, I have implemented a simple version that renders only after the packages are loaded in `VideoPlayer`. 

Would that be fine?

**Comment by allenwq** [comment]
@yichenshen That looks good! Thanks!

**Pull request merged**